crates/egui/src/containers/area.rs

```diff
@@ -209,24 +209,31 @@ impl Area {
     ///
     /// If not set, this will default to `Sense::drag()` if movable, `Sense::click()` if interactable, and `Sense::hover()` otherwise.
     #[inline]
-    pub fn sense(mut self, sense: Sense) -> Self {
+    pub const fn sense(mut self, sense: Sense) -> Self {
         self.sense = Some(sense);
         self
     }
 
     /// `order(Order::Foreground)` for an Area that should always be on top
     #[inline]
-    pub fn order(mut self, order: Order) -> Self {
+    pub const fn order(mut self, order: Order) -> Self {
         self.order = order;
         self
     }
 
     #[inline]
-    pub fn default_pos(mut self, default_pos: impl Into<Pos2>) -> Self {
-        self.default_pos = Some(default_pos.into());
+    pub const fn const_default_pos(mut self, default_pos: Pos2) -> Self {
+        self.default_pos = Some(default_pos);
         self
     }
 
+    /// See [`Self::const_default_pos`].
+    #[inline]
+    pub fn default_pos(self, default_pos: impl Into<Pos2>) -> Self {
+        let default_pos = default_pos.into();
+        self.const_default_pos(default_pos)
+    }
```

Comment:
This seems excessive - realistically, why would you want to construct a `const Area`?

Reply:
It does sound funny, but the overall goal of this PR is to at least enable the user to compute certain initial states at compile-time. I presume that the `#[inline]` already does this for optimized builds, but the added bonus feature of `const` is that the user can now store these `Area` configurations inside a `const` or a `static` so that they are _guaranteed_ to be initialized at compile-time (rather than hoping that `#[inline]` would do that for us).

So yes, semantically, a "movable `const` container" is silly. Though I think the `const`-ness is a misnomer for what we're actually enabling here. That is, we're not enforcing an immutable `Area`; rather, we are enabling the compile-time initialization of an `Area`. The `const` keyword is just an unfortunately overloaded term.

Reply:
Adding `const` to some methods is fine, but duplicating methods is not - please revert those changes

Reply:
Some of the trivially `const` methods in the higher-level abstractions rely on the lower-level `const_*` duplicates. For instance, this PR's modification of the `egui::style:Visuals::dark` constructor invokes `Stroke::const_new` in order to be trivially `const`.

https://github.com/emilk/egui/blob/fae5fdd017f23f83bc2d68a546a05870c5b9da57/crates/egui/src/style.rs#L1280

Would it be amenable to keep the `const` alternatives private in the meantime (i.e., until `const` traits stabilize) just so we can keep the powers of `const` available even in the higher-level abstractions without cluttering the public interface with duplicates?